control-plane/gateways/service.go

```diff
@@ -4,23 +4,28 @@
 package gateways
 
 import (
+	"fmt"
+
+	meshv2beta1 "github.com/hashicorp/consul-k8s/control-plane/api/mesh/v2beta1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
-
-	meshv2beta1 "github.com/hashicorp/consul-k8s/control-plane/api/mesh/v2beta1"
 )
 
-const port = int32(443)
-
 func (b *meshGatewayBuilder) Service() *corev1.Service {
 	var (
-		containerConfig *meshv2beta1.GatewayClassContainerConfig
 		portModifier    = int32(0)
+		port            = int32(443)
 		serviceType     = corev1.ServiceType("")
+		containerConfig *meshv2beta1.GatewayClassContainerConfig
 	)
 
 	if b.gcc != nil {
+		fmt.Println("PORT", b.gcc.Spec.Service.Port)
+
+		if b.gcc.Spec.Service.Port != int32(0) {
```

Comment:
I don't think you'll need the `int32(0)` here, the compiler should be smart enough to infer that a `0` is a `int32` (small example https://go.dev/play/p/NsXMC_WN0iI)